### ISSUE TITLE
Failed to execute child process “blueman-applet”

### ISSUE THREAD

**Issue by Vedrillan** [linuxmint/Cinnamon#8202]
```
 * Cinnamon version 4.0.8
 * Distribution - Mint 19.1
 * NVIDIA Corporation GP107M [GeForce GTX 1050 Mobile] nvidia-driver-410
 * 64 bit
 ```

**Issue**
I upgraded from Linux Mint 19 to 19.1 last week and since then I get this notification when booting
![image](https://user-images.githubusercontent.com/2758851/50397227-3399ea00-076f-11e9-8124-f1ee2ff26a94.png)

From the syslog all I can see is a warning about blueman-apple
`Dec 24 09:30:25 wada-laptop cinnamon-session[6925]: WARNING: t+4,15474s: Failed to start app: Unable to start application: Failed to execute child process “blueman-applet” (No such file or directory)`

Initially my laptop was installed with Linux Mint 18.3 and I believe I saw in a release note somewhere a change from blueman to blueberry, so I am pretty sure I just have some leftover from the 18.3 version.

If you guys have any idea how this blueman-applet can be removed? Because it is not showing from the Applets windows (obviously since it cannot be loaded)

**Comment by icarter09** [comment]
@Vedrillan in `/usr/share/cinnamon/applets`....do you see 'blueberry' or 'blueman' as a listed applet?

**Comment by Vedrillan** [comment]
There is only the blueberry applet

```
wada wada-laptop ~
└─ 09:25:58 $ ls -1 /usr/share/cinnamon/applets/
a11y@cinnamon.org
blueberry@cinnamon.org
calendar@cinnamon.org
expo@cinnamon.org
grouped-window-list@cinnamon.org
inhibit@cinnamon.org
keyboard@cinnamon.org
menu@cinnamon.org
network@cinnamon.org
notifications@cinnamon.org
nvidia-prime@cinnamon.org
on-screen-keyboard@cinnamon.org
panel-launchers@cinnamon.org
power@cinnamon.org
recent@cinnamon.org
removable-drives@cinnamon.org
scale@cinnamon.org
separator@cinnamon.org
settings@cinnamon.org
settings-example@cinnamon.org
show-desktop@cinnamon.org
slideshow@cinnamon.org
sound@cinnamon.org
spacer@cinnamon.org
systray@cinnamon.org
trash@cinnamon.org
user@cinnamon.org
window-list@cinnamon.org
windows-quick-list@cinnamon.org
workspace-switcher@cinnamon.org
xrandr@cinnamon.org
```

**Comment by icarter09** [comment]
@Vedrillan that's a positive sign. If you do a search for 'blueman' or 'blueman-applet' (`grep -rin 'blueman-applet` in `/usr/share/cinnamon` or `/usr/share`), does that return any results?

**Comment by Vedrillan** [comment]
`grep -rin blueman-applet /usr/share/` gives no result.

However `grep -rins blueman /usr/share/` gave a few results
```
Binary file /usr/share/icons/Mint-X-Dark/icon-theme.cache matches
Binary file /usr/share/icons/Mint-X/icon-theme.cache matches
Binary file /usr/share/icons/Mint-Y/icon-theme.cache matches
Binary file /usr/share/icons/hicolor/192x192/status/blueberry.png matches
Binary file /usr/share/icons/hicolor/192x192/status/blueberry-active.png matches
Binary file /usr/share/icons/hicolor/192x192/status/blueberry-disabled.png matches
/usr/share/icons/hicolor/scalable/status/blueberry-disabled.svg:19:   sodipodi:docname="blueman-tray-disabled.svg"
/usr/share/icons/hicolor/scalable/status/blueberry-active.svg:19:   sodipodi:docname="blueman-tray-active.svg"
Binary file /usr/share/icons/hicolor/32x32/status/blueberry.png matches
Binary file /usr/share/icons/hicolor/32x32/status/blueberry-active.png matches
Binary file /usr/share/icons/hicolor/32x32/status/blueberry-disabled.png matches
Binary file /usr/share/icons/hicolor/64x64/status/blueberry.png matches
Binary file /usr/share/icons/hicolor/64x64/status/blueberry-active.png matches
Binary file /usr/share/icons/hicolor/64x64/status/blueberry-disabled.png matches
Binary file /usr/share/icons/hicolor/256x256/status/blueberry.png matches
Binary file /usr/share/icons/hicolor/256x256/status/blueberry-active.png matches
Binary file /usr/share/icons/hicolor/256x256/status/blueberry-disabled.png matches
Binary file /usr/share/icons/hicolor/48x48/status/blueberry.png matches
Binary file /usr/share/icons/hicolor/48x48/status/blueberry-active.png matches
Binary file /usr/share/icons/hicolor/48x48/status/blueberry-disabled.png matches
Binary file /usr/share/icons/hicolor/24x24/status/blueberry.png matches
Binary file /usr/share/icons/hicolor/24x24/status/blueberry-active.png matches
Binary file /usr/share/icons/hicolor/24x24/status/blueberry-disabled.png matches
Binary file /usr/share/icons/hicolor/22x22/status/blueberry.png matches
Binary file /usr/share/icons/hicolor/22x22/status/blueberry-active.png matches
Binary file /usr/share/icons/hicolor/22x22/status/blueberry-disabled.png matches
Binary file /usr/share/icons/hicolor/72x72/status/blueberry.png matches
Binary file /usr/share/icons/hicolor/72x72/status/blueberry-active.png matches
Binary file /usr/share/icons/hicolor/72x72/status/blueberry-disabled.png matches
Binary file /usr/share/icons/hicolor/96x96/status/blueberry.png matches
Binary file /usr/share/icons/hicolor/96x96/status/blueberry-active.png matches
Binary file /usr/share/icons/hicolor/96x96/status/blueberry-disabled.png matches
Binary file /usr/share/app-install/icon-theme.cache matches
/usr/share/app-install/desktop/blueman:blueman-manager.desktop:2:X-AppInstall-Package=blueman
/usr/share/app-install/desktop/blueman:blueman-manager.desktop:7:Comment=Blueman Bluetooth Manager
/usr/share/app-install/desktop/blueman:blueman-manager.desktop:8:Icon=blueman
/usr/share/app-install/desktop/blueman:blueman-manager.desktop:9:Exec=blueman-manager
/usr/share/app-install/desktop/blueman:blueman-adapters.desktop:2:X-AppInstall-Package=blueman
/usr/share/app-install/desktop/blueman:blueman-adapters.desktop:8:Exec=blueman-adapters
/usr/share/app-install/desktop/blueman:blueman-adapters.desktop:9:Icon=blueman-device
Binary file /usr/share/command-not-found/commands.db matches`
```

**Comment by icarter09** [comment]
@Vedrillan what does `ls ~/.local/share/cinnamon/applets` produce?

**Comment by Vedrillan** [comment]
Only 3 results
* cryptocoin@guantanamoe
* multicore-sys-monitor@ccadeptic23
* nvidiaprime@pdcurtis

And still no blueman thing :p

**Comment by icarter09** [comment]
@Vedrillan i just installed those 3 applets and they all seem to work fine. Not sure why it's confused on an applet that is no longer present on your install. Do you have any desklets installed?

**Comment by Vedrillan** [comment]
3 desklets:
* clock@cinnamon.org
* launcher@cinnamon.org
* photoframe@cinnamon.org

I am as lost as you are, I tried to read the code a little and found a comment in the code when this error is generated that says `Silently fail to load xlets that aren't actually installed - but no error, since the user can't do anything about it anyhow`, so it seems it can happen that xlets are loaded even if not installed.

I tried to understand how xlets are listed but I am not familiar enough with the project and haven't found yet :x

**Comment by jaszhix** [comment]
blueman-applet is not a Cinnamon applet. Maybe check startup apps and see if blueman is being loaded, the warning can be suppressed by installing the blueman package, but that shouldn't be necessary. The applet load failure notification is something else, and we would only know the cause unless you shared your ~/.xsession-errors (mentioned in the issue template) or the output of `cinnamon --replace.`

**Comment by Vedrillan** [comment]
Here is the [~/.xsession-errors](https://github.com/linuxmint/Cinnamon/files/2720507/xsession-errors.log)

I indeed had a "Blueman Applet" in the Startup Applications, I do not recall installing it but disabling it removed the warning I showed in my first comment.

But I still have the notifications that says "Problem during Cinnamon startup blabla", if I read the log correctly it seems to be an issue with the multicore-sys-monitor applet, I will disable it and reboot to see.

**Comment by jaszhix** [comment]
Ok, that is `multicore-sys-monitor@ccadeptic23` not starting, an applet I maintain. You just need to update the applet in Settings -> Applets.

**Comment by Vedrillan** [comment]
Ok I updated the multicore-sys-monitor applet and there is no more error notification at bootup now.

I tried a fresh install of mint 18.3 and 19.0 in a VM and saw no Blueman Applet, so on my laptop it must have been my doing somehow.

In the end there is no issue with Cinnamon, maybe a slight improvement by showing something on the main applets view when an applet is out of date? Because I admit that I never thought of going to the `Download` tab to update applets, but I am not a big user of applets either.

Thank you both for your support :)
